packages/rollup-contracts/contracts/ovm/PartialStateManager.sol

```diff
@@ -0,0 +1,329 @@
+    }
+
+    /**
+     * @notice Set the nonce for a particular OVM contract
+     * @param _ovmContractAddress The contract we're setting the nonce of.
+     * @param _value The new nonce.
+     */
+    function setOvmContractNonce(address _ovmContractAddress, uint _value) public {
+        require(msg.sender == address(executionManager));
+        ensureVerifiedContract(_ovmContractAddress);
+
+        // Add this contract to the list of updated contracts
+        updatedContracts[updatedContractsCounter] = _ovmContractAddress;
+        updatedContractsCounter += 1;
+
+        // Return the nonce
+        ovmContractNonces[_ovmContractAddress] = _value;
+    }
+
+    /**
+     * @notice Increment the nonce for a particular OVM contract.
+     * @param _ovmContractAddress The contract we're incrementing by 1 the nonce of.
+     */
+    function incrementOvmContractNonce(address _ovmContractAddress) public {
+        require(msg.sender == address(executionManager));
+        ensureVerifiedContract(_ovmContractAddress);
+
+        // Add this contract to the list of updated contracts
+        updatedContracts[updatedContractsCounter] = _ovmContractAddress;
+        updatedContractsCounter += 1;
+
+        // Increment the nonce
+        ovmContractNonces[_ovmContractAddress] += 1;
+    }
+
+
+    /*****************
+    * Contract Codes *
+    *****************/
+    // This is used when CALLing a contract
+
+    /**
+     * @notice Attaches some code contract to the desired OVM contract. This allows the Execution Manager
+     *         to later on get the code contract address to perform calls for this OVM contract.
+     * @param _ovmContractAddress The address of the OVM contract we'd like to associate with some code.
+     * @param _codeContractAddress The address of the code contract that's been deployed.
+     */
+    function associateCodeContract(address _ovmContractAddress, address _codeContractAddress) public {
+        require(msg.sender == address(executionManager));
+
+        ovmCodeContracts[_ovmContractAddress] = _codeContractAddress;
+    }
+
+    /**
+     * @notice Lookup the code contract for some OVM contract, allowing CALL opcodes to be performed.
+     * @param _ovmContractAddress The address of the OVM contract.
+     * @return The associated code contract address.
+     */
+    function getCodeContractAddress(address _ovmContractAddress) public returns(address) {
+        ensureVerifiedContract(_ovmContractAddress);
+
+        return ovmCodeContracts[_ovmContractAddress];
+    }
+
+    /**
+     * @notice Get the bytecode at some contract address. NOTE: This is code taken from Solidity docs here:
+     *         https://solidity.readthedocs.io/en/v0.5.0/assembly.html#example
+     * @param _codeContractAddress The address of the code contract.
+     * @return The bytecode at this address.
+     */
+    function getCodeContractBytecode(address _codeContractAddress) public view returns (bytes memory codeContractBytecode) {
```

Comment:
make sure we verify the code contract bytecode even if it doesn't exist

Reply:
Added another function `getOvmContractBytecode(...)` which will flag invalid state access